# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/u2f/U2F.cpp
# Commit: 1e35279977a4
# Full Hash: 1e35279977a4b2638e9c2c5f080e99f3afaf3aac
# Author: J.C. Jones <jjones@mozilla.com>
# Date: 2019-04-03 04:07:39
# Description:
#   Bug 1541085 - Web Authentication - Only reset mTransaction on cycle collection r=keeler
#   
#   This stack is pretty clear that calling StopListeningForVisibilityEvents
#   (via ClearTransaction) is a no-go from the cycle collector. We need to instead
#   just do the minimum version of bug 1540378, just reset mTransaction and move on.
# ==============================================================================

diff -r b052410b6673 -r 1e35279977a4 dom/u2f/U2F.cpp
--- a/dom/u2f/U2F.cpp	Tue Apr 02 16:56:41 2019 +0000
+++ b/dom/u2f/U2F.cpp	Tue Apr 02 17:56:47 2019 +0000
@@ -55,7 +55,7 @@
 NS_IMPL_CYCLE_COLLECTION_UNLINK_BEGIN_INHERITED(U2F, WebAuthnManagerBase)
   NS_IMPL_CYCLE_COLLECTION_UNLINK(mTransaction)
   NS_IMPL_CYCLE_COLLECTION_UNLINK_PRESERVED_WRAPPER
-  tmp->ClearTransaction();
+  tmp->mTransaction.reset();
 NS_IMPL_CYCLE_COLLECTION_UNLINK_END
 NS_IMPL_CYCLE_COLLECTION_TRAVERSE_BEGIN_INHERITED(U2F, WebAuthnManagerBase)
   NS_IMPL_CYCLE_COLLECTION_TRAVERSE(mTransaction)